generators/app/index.js

```diff
@@ -174,13 +174,16 @@ module.exports = Generator.extend({
       // Allow distros to modify the composer.json.
       if (typeof options.drupalDistro.modifyComposer == 'function') {
         var done = this.async();
-        var self = this;
-        options.drupalDistro.modifyComposer(self, options, this.composer, isNewProject, done, function(err, result, done) {
+        options.drupalDistro.modifyComposer(options, this.composer, isNewProject, done, function(err, result, done) {
           if (!err && result) {
             this.composer = result;
           }
+          else {
+            this.log.warning("Could not retrieve Octane's composer.json: "  + err);
```

Comment:
As an aside, I moved the logging of the error out of the distro plugin, and so was able to remove passing yo in as a parameter.